lib/fizzy/execute.cpp

```diff
@@ -196,6 +196,50 @@ Instance instantiate(const Module& module)
     return instance;
 }
 
+namespace
+{
+template <typename DstT, typename SrcT>
+inline DstT extend(SrcT in)
+{
+    if constexpr (std::is_signed<SrcT>::value)
+    {
+        using SignedDstT = typename std::make_signed<DstT>::type;
+        return static_cast<DstT>(SignedDstT{in});
+    }
+    else
+        return DstT{in};
+}
+
+template <typename DstT, typename SrcT = DstT>
+inline bool load_from_memory(bytes_view memory, Stack<uint64_t>& stack, const uint8_t* immediates)
+{
+    const auto address = static_cast<uint32_t>(stack.pop());
+    // NOTE: alignment is dropped by the parser
+    const auto offset = read<uint32_t>(immediates);
+    if ((address + offset + sizeof(SrcT)) > memory.size())
+        return false;
+
+    const auto ret = load<SrcT>(memory, address + offset);
+    stack.push(extend<DstT>(ret));
+    return true;
+}
+
+template <typename DstT>
+inline bool store_into_memory(Instance& instance, Stack<uint64_t>& stack, const uint8_t* immediates)
```

Comment:
Here, too, `bytes& memory` for symmetry instead of `instance`

Reply:
Damn! Too fast. Sorry. Fixed.